tests/e2e/utils/TokenHandler.ts

```diff
@@ -0,0 +1,41 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+import axios from 'axios';
+import querystring from 'querystring';
+import { injectable } from 'inversify';
+import { TestConstants } from '../TestConstants';
+
+@injectable()
+export class TokenHandler {
```

Comment:
It also would be great to make the ```TokenHandler``` optional (ITokenHandler).
This will make it possible to fully manage queries with inversify and not think about rebuilding logic in the future.
Because for now, it looks like a bottleneck in the queries handling logic.

Reply:
I'm not sure what you mean by this. Could you please elaborate a bit more?

Reply:
Here you can read extended description of the rule I mentioned.
Of course, this is not a very serious resource, but if you want
I can find a more serious source.
https://www.geeksforgeeks.org/writing-clean-else-statements

Reply:
I mean that with current changes we may easily configure all parts of the ```RequestHandler``` except token obtaining part. And I see that it may be a problem in the future. For example if we will need test "Che" with external ```keycloak```. And this is only one example.
  In this case we will not be able to reconfigure ```RequestHandler``` for successful queries 
using inversify.
  And I suggest to lay this opportunity.
  Besides, it doesn't require a lot of changes, minutes on this stage may save hours or even days in the future.

Reply:
Hmm.. Hosted che is not using keycloak.. So the logic for obtaining active token there is completely different. 
@Katka92 How did you plan to deal with that on hosted che side? Did you want to create your own implementation of `IHeaderHandler` and there use your own logic (possible own class, as it's here) for token obtaining?
I wouldn't even be against in having token obtaining part (what is now `TokenHandler`) implemented direclty inside `MultiUserHeaderHandler`.

Reply:
@rhopp I was for having the code from `TokenHandler` directly in `MultiUserHeaderHandler`, but as @Ohrimenko1988 wanted it strictly separated I created this class. 
When I was creating it, I was thinking about having class (e.g. `HostedCheHeaderHandler`) implementing `IHeaderHandler` which will use our own token handling logic.

Reply:
> Hmm.. Hosted che is not using keycloak.. So the logic for obtaining active token there is completely different.
> @Katka92 How did you plan to deal with that on hosted che side? Did you want to create your own implementation of `IHeaderHandler` and there use your own logic (possible own class, as it's here) for token obtaining?
> I wouldn't even be against having token obtaining part (what is now `TokenHandler`) implemented direclty inside `MultiUserHeaderHandler`.

Exactly, using my suggested way we will don't care about way to obtaining token, because we will be able to veride it as we want, in one place, as simple as possible, without duplication. 
In this case for any che you will need only change token obtaining logic.
Nothing more.

Reply:
If you don't agree you may post your example and I'll post my, and we will compare what is better.

Reply:
In addition, here are "SOLID" principles description:
(which includes "single class responcibility" principle )
https://medium.com/@severinperez/writing-flexible-code-with-the-single-responsibility-principle-b71c4f3f883f

Reply:
@Ohrimenko1988 Here's my suggestion: https://github.com/eclipse/che/compare/master...rhopp:pr/Katka92/15892?expand=1

Reply:
```
 export interface ITokenHandler{

    // Returns either empty object, or object containing Auth header.
    getAuthHeader(): Promise<any>
 } 
```
Interface has name "Token Handler" but returns header. For me this is a case when name doesn't reflect functionality and this is may be tricky.

Reply:
I agree the name was not properly chosen. Other than this... have you found something to disagree in a way how it's architectured? (there can be multiple improper names, forgotten asyncs in my changes... I would like to verify with you, that architecturally it's acceptable and then we can work on these minor issues & properly test it)

Reply:
Here is my variant.
I just suggest stop invent a bicycle and just do things described in the inversify docs
https://github.com/eclipse/che/compare/master...Ohrimenko1988:iokhrime-suggestions?expand=1

Reply:
@Ohrimenko1988 If I compare your solution to mine, we only differ in the fact, that you put "another layer"  there. I didn't... just because I don't find this necessary.

Thinking about it now, I can think of only one reason to keep the mechanism of obtaining token separate - And that's if somebody would need to use the token without using `CheApiRequestHandler`... Is that why you separated this into own layer? If we have good (future) use case for this, let's go with this separation. If no, I would prefer to use "simpler" approach.

Reply:
We have several different places.  My variant gives us huge flexibility and the possibility to reuse all separated modules (for example "Token") in other places. 
    Let's agree to stop discussing about placing all to the one place, at least this approach doesn't respect SOLID principles, and I don't see any reason for intentional misconduct these principles and reducing the quality of the code.
   I strongly disagree with your point that if the proposed solution resolves a specific problem we don't care about anything else. And for forwarding the best practices we need additional reasons.
   Just resolve a problem don't enough for me. We should think strategically. This solution should respect the code quality, clean code principles and don't reduce tests flexibility.

Reply:
ok. Let's go with your variant. 
I don't particularly like the (in my opinion unecessary) overengineering, but I see I don't have any strong case against (except of simplicity).
@Katka92 Can you please act on this and provide update when you have time to do so? Or should somebody else take this responsibility?

Reply:
@Ohrimenko1988 Could you please make a PR with your branch to mine? It will be easier for me to apply all the changes. I'll test it and commit changes here for the final review. Thank you

Reply:
This is not overengineering, just respecting best practices.